core/ipc/grpc/common/src/main/proto/ipc.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.opennms.core.ipc.grpc.common";
+option java_outer_classname = "MinionIpc";
+
+// service definitions of IPC between Minion and OpenNMS
+service OpenNMSIpc {
+    // Streams RPC messages between OpenNMS and Minion.
+    rpc RpcStreaming (stream RpcResponseProto) returns (stream RpcRequestProto) {
+    }
+    // Streams Sink messages from Minion to OpenNMS
+    rpc SinkStreaming (stream SinkMessage) returns (Empty) {
+    }
+}
+
+message Empty {
+
+}
+
+
+message RpcRequestProto {
+    string rpc_id = 1;
+    bytes rpc_content = 2;
+    string system_id = 3;
+    string location = 4;
+    string module_id = 5;
+    uint64 expiration_time = 6;
```

Comment:
Unit of time?

Reply:
Didn't understand.  `uint64` from proto maps to long in java

Reply:
Looking at this field it's unclear if this is seconds, milliseconds, a relative duration or a timestamp.

Reply:
field name `expiration_time` seems reasonable. Added comment to specify that it is ` absolute time elapsed since the epoch in msec`

Reply:
That works too. Thanks.